### ISSUE TITLE
Unable to decrypt

### ISSUE THREAD

**Issue by EqualKirby** [apprenticeharper/DeDRM_tools#78]
I cannot copy the text from console since every time I tried to do that then it crashed......

Here is the snapshot:
https://mdkkuq.bn1304.livefilestore.com/y3pj8PYCtXAoi4AfSD-IiauZqZ6MlSHhGvb6kRf2ANtqLlKaQfMEZSyyEZ4PVCE2vfLTDoDtMdA3gwMLIvmSnGe9F-oIPeLTxAqs5ek4kQPp9dXeUloWy0-r2QGMhJsHmFzoaju2_g6tmrh4V8lB1Sbn_HBM6TZvrFBllrswF1_lJo/error.png?psid=1

It seems that it couldn't find the proper key to remove the drm?
I have uninstalled Kindle for PC and calibre, but it still doesn't work for me.

My Windows is running in Mandarin, and the ebook is in Japanese.
Dose that means I need to change my OS's locale setting?
Or it just warns me that console output cannot display correctly since the font problem?

Hope to find a solution.
Thank you.

**Comment by ElleKayEm** [comment]
It seems that there's been a recent change to the Kindle key file.  Quite a few people have posted logs where it shows that the key can't be retrieved.

**Comment by frostyandy2k** [comment]
I have the same issue so yes, the key file must have moved. I will just post my log, maybe it helps (running on a mac).

Any ideas where it might be? Hopefully the fix is just changing the path.

Thank you alot for this tool !!!

```
shellcommand: /usr/local/bin/python '/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/adobekey.py' '/Users/username/Library/Preferences/com.apprenticealf.dedrm/Local Adept Key.der' > '/Users/username/Library/Preferences/com.apprenticealf.dedrm/DeDRM_temp.log'
shellresult: adobekey.py v6.0
Copyright © 2009-2013 i♥cabbages and Apprentice Alf
 Traceback (most recent call last):
  File "/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/adobekey.py", line 602, in <module>
    sys.exit(cli_main())
  File "/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/adobekey.py", line 525, in cli_main
    keys = adeptkeys()
  File "/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/adobekey.py", line 446, in adeptkeys
    raise ADEPTError("Could not find ADE activation.dat file.")
__main__.ADEPTError: Could not find ADE activation.dat file.
shellcommand: /usr/local/bin/python '/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/kindlekey.py' '/Users/username/Library/Preferences/com.apprenticealf.dedrm/Local Kindle Key.k4i' > '/Users/username/Library/Preferences/com.apprenticealf.dedrm/DeDRM_temp.log'
shellresult: kindlekey.py v1.9
Copyright © 2010-2013 some_updates and Apprentice Alf
Found k4Mac kinf2011 file: /Users/username/Library/Containers/com.amazon.Kindle/Data/Library/Application Support/Kindle/storage/.kinf2011
Couldn't decrypt file.
Could not retrieve Kindle for Mac/PC key.
 
shellcommand: /usr/local/bin/python '/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/ignoblekey.py' '/Users/username/Library/Preferences/com.apprenticealf.dedrm/Nook Study Key.b64' > '/Users/username/Library/Preferences/com.apprenticealf.dedrm/DeDRM_temp.log'
shellresult: ignoblekey.py v1.1
Copyright © 2015 Apprentice Alf
No nook Study log files have been found.
Could not retrieve nook Study key.
 
There is 1 ebook to be decrypted.
Processing file: B00HG5HO92_EBOK.azw
shellcommand: /usr/local/bin/python '/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/k4mobidedrm.py' -k '/Users/username/Library/Preferences/com.apprenticealf.dedrm/Local Kindle Key.k4i' '/Users/username/Documents/Kindle/B00HG5HO92_EBOK.azw' '/Users/username/Desktop/' > '/Users/username/Library/Preferences/com.apprenticealf.dedrm/DeDRM_temp.log'
shellresult: Using Library AlfCrypto DLL/DYLIB/SO
K4MobiDeDrm v5.3.
Copyright © 2008-2013 The Dark Reverser et al.
Error getting database from file /Users/username/Library/Preferences/com.apprenticealf.dedrm/Local Kindle Key.k4i: [Errno 2] No such file or directory: u'/Users/username/Library/Preferences/com.apprenticealf.dedrm/Local Kindle Key.k4i'
MobiDeDrm v0.41.
Copyright © 2008-2012 The Dark Reverser et al.
MOBI header version 8, header length 264
Extra Data Flags: 3
Decrypting Kindle Format 8 ebook: How Would You Move Mount Fuji?: Microsoft's Cult of the Puzzle - How the World's Smartest Companies Select the Most Creative Thinkers
Found 0 keys to try after 0.0 seconds
Crypto Type is: 2
Error decrypting book after 0.0 seconds: No key found in 0 keys tried.
 Traceback (most recent call last):
  File "/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/k4mobidedrm.py", line 238, in decryptBook
    with open(dbfile, 'r') as keyfilein:
IOError: [Errno 2] No such file or directory: u'/Users/username/Library/Preferences/com.apprenticealf.dedrm/Local Kindle Key.k4i'
Traceback (most recent call last):
  File "/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/k4mobidedrm.py", line 248, in decryptBook
    book = GetDecryptedBook(infile, kDatabases, androidFiles, serials, pids, starttime)
  File "/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/k4mobidedrm.py", line 222, in GetDecryptedBook
    mb.processBook(totalpids)
  File "/Users/username/Downloads/DeDRM_tools_6.3.6a/DeDRM_Macintosh_Application/DeDRM.app/Contents/Resources/mobidedrm.py", line 469, in processBook
    raise DrmException(u"No key found in {0:d} keys tried.".format(len(goodpids)))
DrmException: No key found in 0 keys tried.
There was an error with 1 ebook:
B00HG5HO92_EBOK.azw couldn't be decrypted
```

**Comment by ElleKayEm** [comment]
It looks like it's finding it in the same place as usual, but cannot decrypt it in new setups.  See this part:

> Found k4Mac kinf2011 file: /Users/username/Library/Containers/com.amazon.Kindle/Data/Library/Application Support/Kindle/storage/.kinf2011
Couldn't decrypt file.
Could not retrieve Kindle for Mac/PC key.

Same thing in the previous log and others I've seen over at the blog.

**Comment by EqualKirby** [comment]
Does it means we have to wait the update?
Hope the developer may solve this issue easily.

Thank you.

**Comment by frostyandy2k** [comment]
What you could do as workaround (provided you have a physical Kindle device) is use the Kindle decryption. You need to register the serial number of your kindle with the dedrm and then you need to download the file from amazon (as .mobi). The last step is using the software.

I believe the developer is on it, we should give him some time. It seems the key changed location only recently.

**Comment by apprenticeharper** [comment]
I believe that this issue is fixed with DeDRM 6.4.0
